### ISSUE TITLE
Segfault in Github actions

### ISSUE THREAD

**Issue by timurguvenkaya** [oven-sh/bun#6199]
### What version of Bun is running?

v1.0.4 (c21be13b). I also tried with "latest"

### What platform is your computer?

Ubuntu 22.04 (ubuntu latest)

### What steps can reproduce the bug?

Simply use https://github.com/oven-sh/setup-bun and do bun install 

### What is the expected behavior?

Should install dependencies 

### What do you see instead?

<img width="986" alt="Screenshot 2023-09-30 at 16 00 17" src="https://github.com/oven-sh/bun/assets/15787480/900edebe-6830-4e09-8886-4adb31837c17">

**Comment by ikhedkar-swi** [comment]
Same here with `Bun v1.0.3`
```
bun install v1.0.3 (25e69c71)
/runner/_work/_temp/63182e49-c86a-4d67-ab6a-6d0f8112b5cb.sh: line 10:   817 Segmentation fault      (core dumped) bun install
Error: Process completed with exit code 139.
```

Strangely when we ran build using GitHub action without `bun.lockb` we didn't see `Segmentation fault` error.

**Comment by timurguvenkaya** [comment]
So @ikhedkar-swi removing bun.lockdb is working workaround right now? If it is I will do that for now

**Comment by ikhedkar-swi** [comment]
@timurguvenkaya - Yes, for now removing `bun.lockdb` is working fine.

**Comment by timurguvenkaya** [comment]
@ikhedkar-swi Have you removed it completely from the repo or used paths-ignore?

 paths-ignore:
      - "bun.lockb"
      
On my side issue still persists even with 1.0.5

**Comment by Jarred-Sumner** [comment]
Can you share the package.json or bun.lockb? It sounds like a data corruption issue + missing validation.

**Comment by timurguvenkaya** [comment]
@Jarred-Sumner 
Here you go
https://filebin.net/zz1qq2mdwherewle

I also tried deleting the bun lock file and reinstalling with bun install but got this error on many packages

```
bun install v1.0.5 (1bf28e0d)
[40.33ms] migrated lockfile from package-lock.json
  📦 Installing [374/5042] error: FileNotFound installing @aws-sdk/client-ssm
error: FileNotFound installing @haftahave/serverless-ses-template
error: FileNotFound installing @ns3/nx-serverless
error: FileNotFound installing @types/pngjs
error: FileNotFound installing esbuild
error: FileNotFound installing serverless-lift
error: FileNotFound installing serverless-localstack
error: FileNotFound installing pngjs
error: FileNotFound installing @aws-sdk/client-sts
error: FileNotFound installing @aws-sdk/credential-provider-node
error: FileNotFound installing uuid
error: FileNotFound installing cli-table
error: FileNotFound installing babel-plugin-macros
error: FileNotFound installing acorn
error: FileNotFound installing @esbuild/darwin-arm64
error: FileNotFound installing ajv
error: FileNotFound installing @aws-cdk/aws-apigatewayv2-alpha
error: FileNotFound installing change-case
error: FileNotFound installing cidr-split
error: FileNotFound installing constructs
error: FileNotFound installing inquirer
error: FileNotFound installing ora
error: FileNotFound installing stripe
error: FileNotFound installing toml
error: FileNotFound installing @types/cypress
error: FileNotFound installing ethers
error: FileNotFound installing @aws-sdk/credential-provider-ini
error: FileNotFound installing @aws-sdk/credential-provider-sso
error: FileNotFound installing colors
error: FileNotFound installing json-schema-traverse
error: FileNotFound installing @aws-cdk/asset-awscli-v1
error: FileNotFound installing @aws-cdk/asset-kubectl-v20
error: FileNotFound installing @aws-cdk/asset-node-proxy-agent-v6
error: FileNotFound installing capital-case
error: FileNotFound installing constant-case
error: FileNotFound installing header-case
error: FileNotFound installing path-case
error: FileNotFound installing sentence-case
error: FileNotFound installing bignumber.js
error: FileNotFound installing aes-js
error: FileNotFound installing @aws-sdk/client-sso
error: FileNotFound installing @types/body-parser
error: FileNotFound installing @types/serve-static
error: FileNotFound installing @types/accepts
error: FileNotFound installing @types/cookies
error: FileNotFound installing upper-case-first
error: FileNotFound installing upper-case
error: FileNotFound installing @types/send
error: FileNotFound installing ajv
error: FileNotFound installing acorn
  📦 Installing [3486/5042]
error: FileNotFound installing acorn
error: FileNotFound installing acorn
error: FileNotFound installing cosmiconfig
error: FileNotFound installing json-schema-traverse
error: FileNotFound installing rxjs
error: FileNotFound installing argparse
error: FileNotFound installing @adraffy/ens-normalize
error: FileNotFound installing @noble/hashes
error: FileNotFound installing tslib
error: FileNotFound installing ws
error: FileNotFound installing acorn
error: FileNotFound installing tslib
error: FileNotFound installing acorn
error: FileNotFound installing acorn
error: FileNotFound installing acorn
 + @upstash/redis@1.23.3
 + tslib@2.6.2

 2197 packages installed [12.94s]
Failed to install 66 packages
```

**Comment by Jarred-Sumner** [comment]
@timurguvenkaya we changed how some of the install code works in Bun v1.0.15. Can you let us know if you're still running into this?

**Comment by rafaelcorreiapoli** [comment]
Im still running into this on 1.0.20 and 1.0.21

**Comment by Jarred-Sumner** [comment]
I'm pretty confident OP's issue was fixed in Bun v1.0.15, and we've made a huge number of stability improvements since then.

If you still run into something like this, please file a new issue and include the crash report url.
